src/test/modelconfig_test.cpp

```diff
@@ -221,6 +221,98 @@ TEST(ModelConfig, parseShapeParam) {
     EXPECT_EQ(status, ovms::StatusCode::SHAPE_WRONG_FORMAT);
 }
 
+TEST(ModelConfig, parseShapeDynamicParam) {
+    using namespace ovms;
+    ovms::ModelConfig config;
+    // Valid
+    std::string valid_str1 = "(64:128,128,256:512,300:301)";
+    std::string valid_str2 = "{\"input\": \"(1, 3:6, 3, 200:100000)\"}";
+    std::string valid_str3 = "{\"input\": \"auto\", \"extra_input\": \"(10:20)\"}";
+
+    ASSERT_EQ(config.parseShapeParameter(valid_str1), StatusCode::OK);
+    auto shapes = config.getShapes_2();
+    EXPECT_EQ(shapes[ovms::ANONYMOUS_INPUT_NAME].shapeMode, ovms::FIXED);
+    EXPECT_EQ(shapes[ovms::ANONYMOUS_INPUT_NAME].shape, (ovms::Shape{{64, 128}, 128, {256, 512}, {300, 301}}));
+
+    auto shape = ovms::Shape{1, 5, 10};
+
+    ASSERT_EQ(config.parseShapeParameter(valid_str2), StatusCode::OK);
+    shapes = config.getShapes_2();
+    EXPECT_EQ(shapes["input"].shapeMode, ovms::FIXED);
+    EXPECT_EQ(shapes["input"].shape, (ovms::Shape{1, {3, 6}, 3, {200, 100000}}));
+
+    ASSERT_EQ(config.parseShapeParameter(valid_str3), StatusCode::OK);
+    shapes = config.getShapes_2();
+    EXPECT_EQ(shapes["input"].shapeMode, ovms::AUTO);
+    EXPECT_EQ(shapes["input"].shape.getSize(), 0);
+    EXPECT_EQ(shapes["extra_input"].shapeMode, ovms::FIXED);
+    EXPECT_EQ(shapes["extra_input"].shape, (ovms::Shape{{10, 20}}));
+
+    // Invalid
+
+    std::vector<std::string> invalid_str{
+        std::string{"[1:50, 300]"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(9:10,,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(:9,20,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(9:,20,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(9-30,20,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(0,20,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(9::30,20,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(90:10,20,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(-90:10,20,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(?,20,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(2.5:3,20,50)\"}"},
+        std::string{"{\"input\": \"auto\", \"extra_input\": \"(1,20,500000000000000000)\"}"},
+    };
+
+    for (std::string str : invalid_str) {
+        auto status = config.parseShapeParameter(str);
```

Comment:
This is minor: if OVMS does fill Status.details field we can do sth like:
```EXPECT(status, StatusCode::EXPECTED) << status.string();```
To give more information if error occurs. Additionally in this case it would be usefull to know which str failed so you could do:
```EXPECT(status, StatusCode::EXPECTED) << str << "Failed with:" << status.string();```

Reply:
We use the details field to inform requester (user who sends predict request) with additional information. I think we do not need to fill the details field here since this can only happen in model manager thread. The status is never returned to requester.

Reply:
This in fact will help debugging the unit tests in the future. Will add details

Reply:
After further investigation it does not make sense. The log will only appear if the test will fail. In case of test fail, it means the parsing succeeded with StatusCode::OK so no log would be displayed anyway.

This apporach is only useful for positive scenario tests.